src/Certify.Tests/Certify.Core.Tests.Unit/AccessControlTests.cs

```diff
@@ -52,174 +52,806 @@ public Task Add<T>(string itemType, T item)
         public Task Update<T>(string itemType, T item)
         {
             var o = item as AccessStoreItem;
-            return Task.FromResult(_store.TryUpdate(o.Id, o, o));
+            _store.TryGetValue(o.Id, out var value);
+            var c = Task.FromResult((T)Convert.ChangeType(value, typeof(T))).Result as AccessStoreItem;
```

Comment:
This seems super complex with all the Task and type change stuff. What problem were you trying to solve?

Reply:
I had copied part of this logic from the Get method, as the TryUpdate() method's third parameter (c) is a copy of the current value in the store for comparison against the second parameter (the value to update to) for the given key. If there is a better way to get a current copy of the value stored in the key for comparison, I'd gladly change it.

In addition, the if logic below it is to allow us to know if the update has failed or succeeded, as the .Result value returned for r was not readable in AccessControl.UpdateSecurityPrinciple() where it is being called. Without this check, the UpdateSecurityPrinciple() method was always returning true regardless of whether an update was actually successfully executed on the MemoryObjectStore.

Reply:
Sure, I've confused you by mixing Task.FromResult in my return (that's a dotnet thing for methods you want to mark as async that don't actually do anything async in that particular implementation). Don't worry about it, I'll have a look sometime after this is merged.